solution/index.js

```diff
@@ -0,0 +1,306 @@
+    listStorage[listEl.closest('section').id][positionInList] =
+    listEl.textContent
+    localStorage.setItem('tasks', JSON.stringify(listStorage))
+  }
+}
+
+// This function saves the list element I'm hovering over to the DOM
+function listELHoveringOver(event) {
+  if (!event.target.closest('.task')) return
+  tasksContainer.hoveringOver = event.target
+  
+  event.target.onmouseout = () => {
+    tasksContainer.hoveringOver = null
+  }
+}
+
+function altKey(event) {
+  if (event.key !== 'Alt') return
+  if (event.type === 'keydown') {
+    event.preventDefault()
+    document.addEventListener('keydown', taskLocationChange)
+  }
+  if (event.type === 'keyup') {
+    document.removeEventListener('keydown', taskLocationChange)
+  }
+}
+
+function taskLocationChange(event) {
+  const hoveringOver = tasksContainer.hoveringOver
+  
+  // the event only performs a task for keys 1, 2, 3.
+  // And only if the mouse is hovering over a list element.
+  if (['1', '2', '3'].indexOf(event.key) === -1 || !hoveringOver) return
+  
+  // chooses a DOM list to transfer to based on the key that was pressed
+  const lists = ['todo', 'in-progress', 'done']
+  const list = tasksContainer.querySelector(`#${lists[event.key - 1]} > ul`)
+  
+  // if the element is already inside the list the function does nothing.
+  if (list.contains(hoveringOver)) return
+  
+  setLocalStorage(hoveringOver, lists[event.key - 1])
+  
+  // change the DOM
+  hoveringOver.remove()
+  list.prepend(hoveringOver)
+}
+
+function searchFilter() {
+  let taskLists = document.querySelectorAll(`#tasks-container > section > ul`)
+  
+  // I unhide all the elements so they do stay hidden from the search before
+  for (let list of taskLists) {
+    ;[...list.children].forEach((elem) => (elem.hidden = false))
+  }
+  
+  // I hide all the elements that do not contain the text in the search bar
+  for (let list of taskLists) {
+    ;[...list.children].forEach((elem) => {
+      if (!elem.textContent.includes(search.value.toLowerCase())) {
+        elem.hidden = true
+      }
+    })
+  }
+}
+
+/* API Load and Save Data */
+function loadOrSave(event) {
+  if (event.target.id === 'load-btn') loadData()
+  if (event.target.id === 'save-btn') saveData()
+}
+
+async function saveData() {
+  // Added spiner so the user sees that API is currently saving their data.
+  tasksContainer.append(spinner)
+
+  const saveResponse = await fetch(
+    'https://json-bins.herokuapp.com/bin/614b0f854021ac0e6c080cdc',
+    {
+      method: 'PUT',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ tasks: listStorage }),
+    }
+  )
+
+  spinner.remove()
+
+  // tells the user the API failed to save their data
+  if (!saveResponse.ok) {
+    alert(
+      `Your information could not be save. Network Response: ${saveResponse.status}`
+    )
+  }
+}
+
+async function loadData() {
```

Comment:
seperate between network logic & ui logic

Reply:
devide & wrap functions into smaller ones

Reply:
Thanks for all the Input. I appreciate it.